### PR TITLE
[DO NOT MERGE] ToplevelWaf Swagger Changes 

### PR DESCRIPTION
**Pull request by venk8**
Previous PRs:
[Link to Previous PR] (https://github.com/Azure/azure-rest-api-specs/pull/5990)

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by venkatsvpr** [comment]
Do not checkin - Have to get signoff from AFD and the portal team.

**Comment by NullMDR** [comment]
@openapibot sdkautomation rebuild

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json [hunk 1]
```diff
@@ -532,6 +561,129 @@
...
+          "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
+        }
+      },
+      "required": [
+        "matchVariable"
+      ],
+      "description": "Allow to exclude some variable satisfy the condition for the WAF check."
+    }
+  },
+  "parameters": {
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json]: Please don't duplicate existing definitions, use `SubscriptionIdParameter` and `ApiVersionParameter` from `network.json`,

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json [hunk 2]
```diff
@@ -295,6 +295,15 @@
             "$ref": "#/definitions/WebApplicationFirewallCustomRule"
           }
         },
+        "managedRules": {
+          "description": "Describes managed rules inside the policy.",
+          "$ref": "#/definitions/ManagedRuleSets"
+        },
+        "exclusions":{
+          "description": "Describes the Exclusions that are applied on the policy.",
+          "$ref": "#/defintions/CrsExclusions"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json]: Typo in `defintions`. Also, you don't really need to move this to definitions. This code would be fine:
```json
"exclusions": {
  "type": "array",
  "items": {
    "$ref": "#/definitions/CrsExclusionEntry"
  },
  "description": "Describes the Exclusions that are applied on the policy."
}
```

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json (hunk 1 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json]: Yeah. Makes sense!

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json (hunk 2 above)
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json]: Addressed!

**Comment by number213** [comment]
@venkatsvpr please double check your commits, there seems to be a lot of unrelated changes. The CI is also failing because it can't resolve definition `ManagedRuleSetsList`

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
`network-july-release` has been merged, please rebase and retarget to `master`

**Comment by yungezz** [comment]
Hi @venkatsvpr  could you pls update this PR?

**Comment by MikhailTryakhov** [comment]
closing due to NRP 2019-08-01 branch is not ready yet. we've asked owner to change when branch would be ready in a month

**Pull request closed without merging**